### PR TITLE
Fea cleanup stream part1

### PR DESCRIPTION
**Pull request by ChuckHastings**
Addresses part of #1538 and #1422 

Breaking the work for these issues into smaller chunks to make it easier to test, review and approve.

### CODE REVIEW

**Comment by seunghwak** [comment]
Is this PR pretty much complete (and just waiting for CI issues to be resolved) or you have more works to do?

I am worried that I will see a crazy number of merge conflicts if I try to merge my DCSR PR with this PR (both will touch many files in common).

If this PR is nearly complete, I can pull this to my PR first to avoid resolving merge conflicts later.

**Comment by ChuckHastings** [comment]
I was going to do no more in this PR unless there were some review comments.  There's a bunch more files to do this with, but I was going to do them all in batches like this.

Just waiting for the CI issues to be resolved to request reviews.

File: cpp/include/cugraph/dendrogram.hpp [hunk 1]
```diff
@@ -27,10 +27,11 @@ class Dendrogram {
  public:
   void add_level(vertex_t first_index,
                  vertex_t num_verts,
-                 cudaStream_t stream,
+                 rmm::cuda_stream_view const &stream_view,
```
Comment (seunghwak) [inline cpp/include/cugraph/dendrogram.hpp:30]: Should we pass a const reference here?

Considering that
1) cuda_stream_view is just a wrapper for cudaStream_t.
https://github.com/rapidsai/rmm/blob/branch-21.08/include/rmm/cuda_stream_view.hpp#L108
2) and cuDF passes cuda_stream_view by value
https://github.com/rapidsai/cudf/blob/branch-21.08/cpp/include/cudf/detail/copy.hpp#L76
https://github.com/rapidsai/cudf/blob/branch-21.08/cpp/include/cudf/detail/groupby.hpp#L47

**Review by seunghwak — APPROVED** [review]
Not something critical, but we may need to review whether should we pass a cuda_stream_view object by value or const reference.

Besides that, looks good to me.

File: cpp/include/cugraph/dendrogram.hpp (hunk 1 above)
Comment (ChuckHastings) [reply cpp/include/cugraph/dendrogram.hpp]: Pushed the change to pass by value.

**Review by aschaffer — APPROVED** [review]

**Comment by BradReesWork** [comment]
@gpucibot merge

**Pull request merged**